src/files/BrsFile.ts

```diff
@@ -1004,6 +1011,195 @@ export class BrsFile {
+        } else if (isNamespaceStatement(statement)) {
+            symbolKind = SymbolKind.Namespace;
+            for (const childStatement of statement.body.statements) {
+                const symbol = this.getDocumentSymbol(childStatement);
+                if (symbol) {
+                    children.push(symbol);
+                }
+            }
+        } else if (isClassStatement(statement)) {
+            symbolKind = SymbolKind.Class;
+            for (const childStatement of statement.body) {
+                const symbol = this.getDocumentSymbol(childStatement);
+                if (symbol) {
+                    children.push(symbol);
+                }
+            }
+        } else {
+            return;
+        }
+
+        const name = statement.getName(ParseMode.BrighterScript);
+        return DocumentSymbol.create(name, '', symbolKind, statement.range, statement.range, children);
+    }
+
+    /**
+     * Builds a single SymbolInformation object for use by LanguageServer's onWorkspaceSymbol functionality
+     */
+    private generateWorkspaceSymbols(statement: Statement, containerStatement?: ClassStatement | NamespaceStatement) {
+        let symbolKind: SymbolKind;
+        const symbols = [];
+
+        if (isFunctionStatement(statement)) {
+            symbolKind = SymbolKind.Function;
+        } else if (isClassMethodStatement(statement)) {
+            symbolKind = SymbolKind.Method;
+        } else if (isNamespaceStatement(statement)) {
+            symbolKind = SymbolKind.Namespace;
+
+            for (const childStatement of statement.body.statements) {
+                for (const symbol of this.generateWorkspaceSymbols(childStatement, statement)) {
+                    symbols.push(symbol);
+                }
+            }
+        } else if (isClassStatement(statement)) {
+            symbolKind = SymbolKind.Class;
+
+            for (const childStatement of statement.body) {
+                for (const symbol of this.generateWorkspaceSymbols(childStatement, statement)) {
+                    symbols.push(symbol);
+                }
+            }
+        } else {
+            return symbols;
+        }
+
+        const name = statement.getName(ParseMode.BrighterScript);
+        const uri = util.pathToUri(this.pathAbsolute);
+        const symbol = SymbolInformation.create(name, symbolKind, statement.range, uri, containerStatement?.getName(ParseMode.BrighterScript));
+        symbols.push(symbol);
+        return symbols;
+    }
+
+    /**
+     * Given a position in a file, if the position is sitting on some type of identifier,
+     * go to the definition of that identifier (where this thing was first defined)
+     */
+    public async getDefinition(position: Position) {
+        await this.isReady();
+
+        let results: Location[] = [];
+
+        //get the token at the position
+        const token = this.getTokenAt(position);
+
+        let definitionTokenTypes = [
```

Comment:
[Certain keywords](https://github.com/rokucommunity/brighterscript/blob/master/src/lexer/TokenKind.ts#L418) are allowed to be used as local variable names, so we should probably include those in this list.

Also, I'm not overly familiar with the full reach of this function, but [some keywords](https://github.com/rokucommunity/brighterscript/blob/07954eed46225aa26408ec00406bf48286f28cb3/src/lexer/TokenKind.ts#L340) are allowed to be object property names, so you _might_ want to include these as well: